### DISCUSSION TITLE
How to switch accounts in Snap?

### DISCUSSION THREAD

**Discussion by enu-kuro** [MetaMask/snaps#1338] [Q&A]
How can I switch the account in a Metamask Snap? 
I changed my connected account from Account1 to Account2, but the provider keeps returning Account1 even though in the browser it returns Account2.

**Comment by enu-kuro** [comment]
I fixed the issue. 
I didn't correctly reconnect to snap.

**Comment by enu-kuro** [comment]
still often fail to switch accounts...

**Comment by ziad-saab** [comment]
You cannot get or switch accounts from a snap, only from a dapp.

**Reply by enu-kuro** [reply]
In a dapp, I switch accounts correctly, but when I call 'eth_requestAccounts' in snap, returned account is always Account1. 
It doesn't  seem be synchronized.
I'm able to switch once before, so probably I missed a necessary step though I don't know what it is. 
```
  const provider = new ethers.providers.Web3Provider(
    ethereum as unknown as BaseProvider,
  );
  const accounts = await provider.send('eth_requestAccounts', []);
  console.log('accounts', accounts);
  ```

**Reply by ritave** [reply]
@enu-kuro snaps uses the same RPC infrastructure as the dApps, so it should behave the same. Can you provide a repository or source code that minimally reproduces this problem so we can take a look at it?
